samples/snippets/cloud-client/src/main/java/privateca/CreateCertificate.java

```diff
@@ -0,0 +1,186 @@
+              // Set the fully qualified domain name.
+              .setSubjectAltName(SubjectAltNames.newBuilder().addDnsNames(domainName).build())
+              .build();
+
+      // Set the X.509 fields required for the certificate.
+      X509Parameters x509Parameters =
+          X509Parameters.newBuilder()
+              .setKeyUsage(
+                  KeyUsage.newBuilder()
+                      .setBaseKeyUsage(
+                          KeyUsageOptions.newBuilder()
+                              .setDigitalSignature(true)
+                              .setKeyEncipherment(true)
+                              .setCertSign(true)
+                              .build())
+                      .setExtendedKeyUsage(
+                          ExtendedKeyUsageOptions.newBuilder().setServerAuth(true).build())
+                      .build())
+              .setCaOptions(CaOptions.newBuilder().setIsCa(true).buildPartial())
+              .build();
+
+      // Create certificate.
+      Certificate certificate =
+          Certificate.newBuilder()
+              .setConfig(
+                  CertificateConfig.newBuilder()
+                      .setPublicKey(publicKey)
+                      .setSubjectConfig(subjectConfig)
+                      .setX509Config(x509Parameters)
+                      .build())
+              .setLifetime(Duration.newBuilder().setSeconds(certificateLifetime).build())
+              .build();
+
+      // Create the Certificate Request.
+      CreateCertificateRequest certificateRequest =
+          CreateCertificateRequest.newBuilder()
+              .setParent(CaPoolName.of(project, location, caPoolName).toString())
+              .setCertificateId(certificateName)
+              .setCertificate(certificate)
+              .setIssuingCertificateAuthorityId(certificateAuthorityName)
+              .build();
+
+      // Get the Certificate response.
+      ApiFuture<Certificate> future =
+          certificateAuthorityServiceClient
+              .createCertificateCallable()
+              .futureCall(certificateRequest);
+
+      // Get the PEM encoded, signed X.509 certificate.
+      // response.getPemCertificate()
+      // To verify the obtained certificate, use this intermediate chain list.
+      // response.getPemCertificateChainList()
+      Certificate response = future.get();
+    }
+  }
+
+  // Get the public Key used for signing the certificate from Cloud KMS.
+  public static ByteString retrievePublicKey(
+      String project, String kmsLocation, String keyRingId, String keyId, String keyVersionId)
+      throws IOException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the `certificateAuthorityServiceClient.close()` method on the client to safely
```

Comment:
should this be `client.close()`? Or will this be closed in this try-with-resources?

Reply:
This would be closed in the try-with-resources. ditto as above. Thanks!

Reply:
SGTM! However, I think `certificateAuthorityServiceClient.close()` should be changed to `client.close()` to match the code snippet below?

Reply:
You would like me to rename 'certificateAuthorityServiceClient' to just 'client', throughout the code snippet?

Reply:
Thanks for the explanation. Fixed it.